flexmeasures/data/models/reporting/tests/test_tibber_reporter.py

```diff
@@ -0,0 +1,267 @@
+
+    electricity_tax = GenericAsset(name="Energy Tax", generic_asset_type=tax)
+
+    tibber_report = GenericAsset(name="TibberReport", generic_asset_type=report)
+
+    db.session.add_all([electricity_price, VAT_asset, electricity_tax, tibber_report])
+
+    # Taxes
+    VAT = Sensor(
+        "VAT",
+        generic_asset=VAT_asset,
+        event_resolution=timedelta(days=365),
+        unit="",
+    )
+    EnergyTax = Sensor(
+        "EnergyTax",
+        generic_asset=electricity_tax,
+        event_resolution=timedelta(days=365),
+        unit="EUR/MWh",
+    )
+
+    # Tibber Tariff
+    tibber_tariff = Sensor(
+        "Tibber Tariff",
+        generic_asset=electricity_price,
+        event_resolution=timedelta(days=365),
+        unit="EUR/MWh",
+    )
+
+    db.session.add_all([VAT, EnergyTax, tibber_tariff])
+
+    """
+        Saving TimeBeliefs to the DB
+    """
+
+    # Add EnergyTax, VAT and Tibber Tariff beliefs to the DB
+    for sensor, source_name, value in [
+        (VAT, "Tax Authority", 0.21),
+        (EnergyTax, "Tax Authority", 125.99),  # EUR / MWh
+        (tibber_tariff, "Tibber", 18.0),  # EUR /MWh
+    ]:
+        belief = TimedBelief(
+            sensor=sensor,
+            source=DataSource(source_name),
+            event_value=value,
+            event_start=datetime(2023, 1, 1, tzinfo=utc),
+            belief_time=datetime(2023, 1, 1, tzinfo=utc),
+        )
+
+        db.session.add(belief)
+
+    # DA Prices
+    entsoe = DataSource("ENTSOE")
+    da_prices = Sensor(
+        "DA prices",
+        generic_asset=electricity_price,
+        event_resolution=timedelta(hours=1),
+    )
+    db.session.add(da_prices)
+    da_prices_beliefs = beliefs_from_timeseries(index, entsoe_prices, da_prices, entsoe)
+    db.session.add_all(da_prices_beliefs)
+
+    tibber_report_sensor = Sensor(
+        "TibberReportSensor",
+        generic_asset=tibber_report,
+        event_resolution=timedelta(hours=1),
+        unit="EUR/MWh",
+    )
+    db.session.add(tibber_report_sensor)
+
+    return tibber_report_sensor
+
+
+def test_tibber_reporter(tibber_test_data):
+    """
+    This test checks if the calculation of the energy prices gets close enough to the ones
+    displayed in Tibber's App.
+    """
+
+    tibber_report_sensor = tibber_test_data
+
+    tibber_reporter = TibberReporter(tibber_report_sensor)
+
+    result = tibber_reporter.compute(
+        start=datetime(2023, 4, 13, tzinfo=utc),
+        end=datetime(2023, 4, 13, 23, tzinfo=utc),
+    )
+
+    # check that we got a result for 24 hours
+    assert len(result) == 24
+
+    tibber_app_price_df = (
+        pd.DataFrame(tibber_app_price, index=index, columns=["event_value"])
+        * 10  # convert cents/kWh to EUR/MWh
+    )
+
+    error = abs(result - tibber_app_price_df)
+
+    # check that (EPEX + EnergyTax + Tibber Tariff)*(1 + VAT) = Tibber App Price
+    assert error.sum()[0] == 0
```

Comment:
I didn't know this, but it turns out that [Pandas silently sums a column with all NaN values to 0 by default](https://stackoverflow.com/questions/33448003/sum-across-all-nans-in-pandas-returns-zero), which is what is happening here.

I suggest to change this line to:
```
assert error.sum(min_count=1).event_value == 0
```
And then to find out when the test started failing. :)

Reply:
good catch! The method `compute` was overriding the values with a `None` and therefore, the resolution wasn't passed to `search_beliefs`